src/main/java/uk/gov/nationalarchives/pdi/step/xml/CanonicalStepDialog.java

```diff
@@ -0,0 +1,283 @@
+                cancel();
+            }
+        };
+        lsOK = new Listener() {
+            public void handleEvent( Event e ) {
+                ok();
+            }
+        };
+        wCancel.addListener( SWT.Selection, lsCancel );
+        wOK.addListener( SWT.Selection, lsOK );
+
+        // default listener (for hitting "enter")
+        lsDef = new SelectionAdapter() {
+            public void widgetDefaultSelected( SelectionEvent e ) {
+                ok();
+            }
+        };
+        wStepname.addSelectionListener( lsDef );
+        //wHelloFieldName.addSelectionListener( lsDef );
+
+        // Detect X or ALT-F4 or something that kills this window and cancel the dialog properly
+        shell.addShellListener( new ShellAdapter() {
+            public void shellClosed( ShellEvent e ) {
+                cancel();
+            }
+        } );
+
+        // Set/Restore the dialog size based on last position on screen
+        // The setSize() method is inherited from BaseStepDialog
+        setSize();
+
+        // populate the dialog with the values from the meta object
+        populateDialog();
+
+        // restore the changed flag to original value, as the modify listeners fire during dialog population
+        meta.setChanged( changed );
+
+        // open dialog and enter event loop
+        shell.open();
+        while ( !shell.isDisposed() ) {
+            if ( !display.readAndDispatch() ) {
+                display.sleep();
+            }
+        }
+
+        // at this point the dialog has closed, so either ok() or cancel() have been executed
+        // The "stepname" variable is inherited from BaseStepDialog
+        return stepname;
+    }
+
+    /**
+     * This helper method puts the step configuration stored in the meta object
+     * and puts it into the dialog controls.
+     */
+    private void populateDialog() {
+        wStepname.selectAll();
+        xmlInputField.setText(meta.getInputField());
+        xmlOutputFieldName.setText( meta.getOutputField() );
+    }
+
+    /**
+     * Called when the user cancels the dialog.
+     */
+    private void cancel() {
+        // The "stepname" variable will be the return value for the open() method.
+        // Setting to null to indicate that dialog was cancelled.
+        stepname = null;
+        // Restoring original "changed" flag on the met aobject
+        meta.setChanged( changed );
+        // close the SWT dialog window
+        dispose();
+    }
+
+    /**
+     * Called when the user confirms the dialog
+     */
+    private void ok() {
+        // The "stepname" variable will be the return value for the open() method.
+        // Setting to step name from the dialog control
+        stepname = wStepname.getText();
+        // Setting the  settings to the meta object
+        meta.setInputField(xmlInputField.getText());
+        meta.setOutputField( xmlOutputFieldName.getText() );
+        // close the SWT dialog window
```

Comment:
I think you need to maybe call `meta.setChanged`? - e.g. https://github.com/nationalarchives/kettle-jena-plugins/blob/main/src/main/java/uk/gov/nationalarchives/pdi/step/jena/serializer/JenaSerializerStepDialog.java#L542